transport-classes-io_uring/src/main/java/io/netty/channel/uring/AbstractIoUringStreamChannel.java

```diff
@@ -335,6 +352,36 @@ protected int scheduleRead0(boolean first, boolean socketIsEmpty) {
             }
         }
 
+        private int scheduleReadProviderBuffer(IoUringBufferRing bufferRing) {
+            short bgId = bufferRing.bufferGroupId();
+            try {
+                int chunkSize = bufferRing.chunkSize();
+                IoUringIoRegistration registration = registration();
+                IoUringBufferRing ioUringBufferRing = ((IoUringIoHandler) registration.ioHandler())
+                        .findBufferRing(bgId);
+
+                if (!ioUringBufferRing.isFull()) {
+                    ioUringBufferRing.appendBuffer(1);
```

Comment:
just make this a single method, at this point - and constant fold giving the method a proper name

Reply:
like this?  Is there anything I didn't understand? What are the advantages of doing it this way?

``` java
    void appendBufferIfNotFull() {
        if (!isFull()) {
            appendBuffer(1);
        }
    }
``

Reply:
You can even avoid having 2 methods as a whole, since they are meant to always been used together (or at least it seems like that right now).
It's just a choice to reduce the API surface , but it's a personal thing - don't be too worried and do as you prefer :+1: